library/entity/interaction/src/main/java/org/quiltmc/qsl/entity/interaction/api/player/UseBlockEvents.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.entity.interaction.api.player;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.quiltmc.qsl.base.api.event.Event;
+
+/**
+ * Contains events related to a Player right-clicking a block.
+ */
+public class UseBlockEvents {
+
+	/**
+	 * A callback that is invoked <strong>before</strong> a {@link PlayerEntity} right-clicks a block.
+	 * <p>
+	 * Implementations should not assume the action has been completed.
+	 * <p>
+	 * Upon return:
+	 * <ul>
+	 *     <li>{@link ActionResult#SUCCESS}/{@link ActionResult#CONSUME}/{@link ActionResult#CONSUME_PARTIAL}
```

Comment:
```suggestion
	 *     <li>{@link ActionResult#SUCCESS}, {@link ActionResult#CONSUME}, or {@link ActionResult#CONSUME_PARTIAL}
```
Alternatively, iirc there is a method in `ActionResult` that checks this for you that you could link instead.